napari/layers/points/_tests/test_points.py

```diff
@@ -445,6 +445,38 @@ def test_properties():
     assert np.all(layer.properties['point_type'] == paste_annotations)
 
 
+@pytest.mark.parametrize("attribute", ['edge', 'face'])
+def test_adding_properties(attribute):
+    """Test adding properties to an existing layer"""
+    shape = (10, 2)
+    np.random.seed(0)
+    data = 20 * np.random.random(shape)
+    layer = Points(data)
+
+    # add properties
+    properties = {'point_type': np.array(['A', 'B'] * int(shape[0] / 2))}
+    layer.properties = properties
+    np.testing.assert_equal(layer.properties, properties)
+
+    # add properties as a dataframe
+    properties_df = pd.DataFrame(properties)
+    layer.properties = properties_df
+    np.testing.assert_equal(layer.properties, properties)
+
+    # add properties as a dictionary with list values
+    properties_list = {'point_type': ['A', 'B'] * int(shape[0] / 2)}
```

Comment:
```suggestion
    properties_list = {'point_type': ['A', 'B'] * (shape[0] // 2)}
```

Having said this, I feel this is a bit brittle, and will only work when `shape[0]` is even. I think the "correct" way here would be to use itertools:

```python
from itertools import cycle, islice

point_type = list(islice(cycle('AB'), None, shape[0]))
```

Or:

```python
from itertools import cycle
import toolz as tz

point_type = list(tz.take(shape[0], cycle('AB')))
```

Reply:
And you make this multiple times in this test so might as well make it once (as a list) and transform it to `np.array` where needed.

Reply:
Thanks for the suggestion. I've implemented it as a function using the islice method. That way we can make property arrays of different lengths, if needed.